packages/remote-feature-flag-controller/src/utils/user-segmentation-utils.ts

```diff
@@ -19,6 +20,43 @@ const MIN_UUID_V4_BIGINT = uuidStringToBigInt(MIN_UUID_V4);
 const MAX_UUID_V4_BIGINT = uuidStringToBigInt(MAX_UUID_V4);
 const UUID_V4_VALUE_RANGE_BIGINT = MAX_UUID_V4_BIGINT - MIN_UUID_V4_BIGINT;
 
+/**
+ * Calculates a deterministic threshold value between 0 and 1 for A/B testing.
+ * This function hashes the user's MetaMetrics ID combined with the feature flag name
+ * to ensure consistent group assignment across sessions while varying across different flags.
+ * Normalizes inputs to lowercase to ensure case-insensitive consistency.
+ *
+ * @param metaMetricsId - The user's MetaMetrics ID (must be non-empty)
+ * @param featureFlagName - The feature flag name to create unique threshold per flag
+ * @returns A promise that resolves to a number between 0 and 1
+ * @throws Error if metaMetricsId is empty
+ */
+export async function calculateThresholdForFlag(
+  metaMetricsId: string,
+  featureFlagName: string,
+): Promise<number> {
+  if (!metaMetricsId) {
+    throw new Error('MetaMetrics ID cannot be empty');
+  }
+
+  // Normalize to lowercase to ensure case-insensitive consistency
+  const normalizedId = metaMetricsId.toLowerCase();
+  const normalizedFlagName = featureFlagName.toLowerCase();
+  const seed = normalizedId + normalizedFlagName;
```

Comment:
Nit: Is this step necessary? Do we expect to get the same ID in alternating mixed/lower case? That would be surprising.

```suggestion
  const normalizedFlagName = featureFlagName.toLowerCase();
  const seed = metaMetricsId + normalizedFlagName;
```